compat/debug.h

```diff
@@ -25,11 +25,13 @@ extern int debug_enabled;
 /* the macro overload style is really a gcc-ism */
 #if defined(__GNUC__) || defined(__SUNPRO_CC)
 
-#define debug(X...) \
-                     if (debug_enabled) { \
-                         fprintf(stderr, "%s(%d): pid=%ld :", __FILE__, __LINE__, static_cast<long>(getpid())); \
-                         fprintf(stderr,X); \
-                     } else (void)0
+#define debug(X...)                                                                                \
+    do {                                                                                           \
+        if (debug_enabled) {                                                                       \
+            fprintf(stderr, "%s(%d): pid=%ld :", __FILE__, __LINE__, static_cast<long>(getpid())); \
+            fprintf(stderr, X);                                                                    \
+        }                                                                                          \
+    } while (/*CONSTCOND*/  0)
```

Comment:
Please copy debugs() style instead of introducing your own:

* Remove beautiful alignment of line continuation backslashes.
* Remove extra space on the `while` condition line.